src/group_theory/free_abelian_group.lean

```diff
@@ -2,19 +2,61 @@
 Copyright (c) 2018 Kenny Lau. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Kenny Lau
-
-Free abelian groups as abelianization of free groups.
-
--- TODO: rewrite in terms of finsupp
 -/
+
 import algebra.group.pi
 import group_theory.free_group
 import group_theory.abelianization
+import algebra.module.basic -- we use the ℤ-module structure on an add_comm_group in punit_equiv
+
+/-!
+# Free abelian groups
+
+The free abelian group on a type `α`, defined as the abelianisation of
+the free group on `α`. The free abelian group on `α` is the left adjoint of the
```

Comment:
Maybe it would be helpful to include a pointer to `algebra/category/adjunctions` where this adjointness is actually provided?

Reply:
Done.